src/Microsoft.IdentityModel.Tokens/Validation/Results/Details/AudienceValidationError.cs

```diff
@@ -40,12 +31,18 @@ public AudienceValidationError(
         internal override Exception GetException()
         {
             if (ExceptionType == typeof(SecurityTokenInvalidAudienceException))
-                return new SecurityTokenInvalidAudienceException(MessageDetail.Message) { InvalidAudience = Utility.SerializeAsSingleCommaDelimitedString(_tokenAudiences) };
+            {
+                var exception = new SecurityTokenInvalidAudienceException(MessageDetail.Message, InnerException) { InvalidAudience = Utility.SerializeAsSingleCommaDelimitedString(TokenAudiences) };
+                exception.SetValidationError(this);
+
+                return exception;
+            }
 
             return base.GetException(ExceptionType, null);
         }
 
-        internal IList<string>? TokenAudiences => _tokenAudiences;
+        protected IList<string>? TokenAudiences { get; set; }
```

Comment:
Do we need a 'set' here and below, if the value is set in the ctor?

Reply:
Removed to force any subclasses to set this via the base constructor.